kyo-core/shared/src/main/scala/kyo/fibers.scala

```diff
@@ -171,24 +172,33 @@ object Fibers extends Joins[Fibers]:
                 }
 
     def race[T](l: Seq[T < Fibers])(using f: Flat[T < Fibers]): T < Fibers =
-        l.size match
-            case 0 => IOs.fail("Can't race an empty list.")
-            case 1 => l(0)
-            case _ =>
-                Fibers.get(raceFiber[T](l))
+        Fibers.get(raceFiber[T](l))
 
     def raceFiber[T](l: Seq[T < Fibers])(using f: Flat[T < Fibers]): Fiber[T] < IOs =
         l.size match
             case 0 => IOs.fail("Can't race an empty list.")
             case 1 => Fibers.run(l(0))
-            case _ =>
+            case size =>
                 Locals.save.map { st =>
                     IOs {
-                        val p = new IOPromise[T]
+                        class State extends AbstractFunction1[T < IOs, Unit]:
```

Comment:
I've added this class to reduce the memory footprint of the method. It avoids the closure allocation in the `foreach` body.